sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADOAuth2AuthorizedClientRepository.java

```diff
@@ -1,139 +0,0 @@
-
-import static com.azure.spring.aad.AADClientRegistrationRepository.AZURE_CLIENT_REGISTRATION_ID;
-
-/**
- * OAuth2AuthorizedClientRepository used for AAD oauth2 clients.
- */
-public class AADOAuth2AuthorizedClientRepository implements OAuth2AuthorizedClientRepository {
-
-    private static final Logger LOGGER = LoggerFactory.getLogger(AADOAuth2AuthorizedClientRepository.class);
-
-    private final AADClientRegistrationRepository repo;
-    private final OAuth2AuthorizedClientRepository delegate;
-    private final OAuth2AuthorizedClientProvider provider;
-
-    public AADOAuth2AuthorizedClientRepository(AADClientRegistrationRepository repo) {
-        this(repo,
-            new JacksonHttpSessionOAuth2AuthorizedClientRepository(),
-            new RefreshTokenOAuth2AuthorizedClientProvider());
-    }
-
-    public AADOAuth2AuthorizedClientRepository(AADClientRegistrationRepository repo,
-                                               OAuth2AuthorizedClientRepository delegate,
-                                               OAuth2AuthorizedClientProvider provider) {
-        this.repo = repo;
-        this.delegate = delegate;
-        this.provider = provider;
-    }
-
-    @Override
-    public void saveAuthorizedClient(OAuth2AuthorizedClient authorizedClient,
-                                     Authentication principal,
-                                     HttpServletRequest request,
-                                     HttpServletResponse response) {
-        // Todo (rujche) Fix OAuth2AuthorizedClient deserializer and memory leakage problem
-        //  1. Fix the deserializer of grant type 'on_behalf_of' when loading OAuth2AuthorizedClient of OBO client from JacksonHttpSessionOAuth2AuthorizedClientRepository.
-        //  2. Fix the memory leakage problem when saving the resource server's OAuth2AuthorizedClient into InMemoryOAuth2AuthorizedClientService.
-        delegate.saveAuthorizedClient(authorizedClient, principal, request, response);
-    }
-
-    @Override
-    @SuppressWarnings("unchecked")
-    public <T extends OAuth2AuthorizedClient> T loadAuthorizedClient(String clientRegistrationId,
-                                                                     Authentication principal,
-                                                                     HttpServletRequest request) {
-        OAuth2AuthorizedClient result = delegate.loadAuthorizedClient(clientRegistrationId, principal, request);
-        if (result != null) {
-            return (T) result;
-        }
-
-        if (repo.isAzureDelegatedClientRegistration(clientRegistrationId)) {
-            OAuth2AuthorizedClient azureClient = loadAuthorizedClient(getAzureClientId(), principal, request);
-            if (azureClient == null) {
-                throw new ClientAuthorizationRequiredException(AZURE_CLIENT_REGISTRATION_ID);
-            }
-            OAuth2AuthorizedClient fakeAuthzClient = createFakeAuthzClient(azureClient, clientRegistrationId, principal);
```

Comment:
For me this looks like an implicit client creation functionality has been removed here .. a "loadOrCreate" so to speak. Other than using the @RegisteredOAuth2AuthorizedClient in controllers how can higher level services now request an authorizedClient which is not the default AZURE client? Thanks in advance

Reply:
Hi, @querwurzel.
Thanks for reaching out.
Please refer to `AADAzureDelegatedOAuth2AuthorizedClientProvider` and `AADAuthorizationGrantType.AZURE_DELEGATED` to get more information.